macros/edr/dbt_artifacts/upload_dbt_tests.sql

```diff
@@ -174,6 +175,40 @@
     {{- return(test_type) -}}
 {%- endmacro -%}
 
+
+{% macro get_test_short_name(node_dict, test_metadata) %}
+    {#
+    If there is a custom name it overrides the dbt auto generated long name.
+    This is a best effort to extract custom names.
+    dbt automated name generation -
+        - Generic test long name starts with the generic name or source_generic
+        - Generic tests from packages long name starts with the package_generic or package_source_generic
+    #}
+
+    {% set generic_test_name = test_metadata.get('name') %} {# 'unique', 'relationships', 'volume_anomalies' etc #}
+    {% set test_package_name = test_metadata.get('namespace') %}
+    {% set specific_test_name = node_dict.get('name') %} {# Test custom name or dbt auto generated long name #}
+    {%- if specific_test_name and generic_test_name and test_package_name %}
```

Comment:
Only `test_package_name` can potentially be `None` here.
Maybe we should change the `if` only to it and move the non-package checks to the `elif` below that can be an `else`.

Roughly (pseudo):

```py
{% if test_package_name %}
  {{ return(generic_test_name if (specific_test_name.startswith(test_package_name + '_' + generic_test_name) or specific_test_name.startswith(test_package_name + '_source_' + generic_test_name))
              else specific_test_name) }}
{% else %}
  {{ return(generic_test_name if (specific_test_name.startswith(generic_test_name) or specific_test_name.startswith( 'source_' + generic_test_name))
              else specific_test_name) }}
{% endif %}
```